black-mulberry-core/src/main/java/com/black/mulberry/core/service/ProductService.java

```diff
@@ -0,0 +1,113 @@
+        InputStream inputStream = new FileInputStream(folderPath + File.separator + fileName);
+        return IOUtils.toByteArray(inputStream);
+    }
+
+    public Optional<Product> getProductById(long id){
+        return productRepository.findById(id);
+    }
+
+    public ProductResponse saveProduct(ProductRequest productRequest) {
+        Optional<User> optionalUser = userRepository.findById(productRequest.getUserId());
+        Optional<CategoryProduct> categoryProduct = categoryProductRepository.findById(productRequest.getCategoryProductId());
+        Product product = Product.builder()
+                .title(productRequest.getTitle())
+                .price(productRequest.getPrice())
+                .stock(productRequest.getStock())
+                .picUrl(productRequest.getPicUrl())
+                .description(productRequest.getDescription())
+                .categoryProduct(categoryProduct.get())
+                .user(optionalUser.get())
+                .build();
+        Product save = productRepository.save(product);
+        return productMapper.toResponse(save);
+    }
+
+
+    public List<ProductResponse> findAllProducts() {
+        List<Product> productList = productRepository.findAllByIsDeleteFalse();
+        return productList.stream()
+                .map(productMapper::toResponse)
+                .collect(Collectors.toCollection(LinkedList::new));
+    }
+
+    public void deleteProduct(long id){
+        Optional<Product> byId = productRepository.findById(id);
+        if (byId.isPresent()){
+            Product product = byId.get();
+            product.setDelete(true);
+            productRepository.save(product);
+        }
+        throw new ProductNotExistException("product with" + id +"don't exist");
+    }
+
+    public ProductResponse findById(long id){
+        Product product = productRepository.findByIdAndIsDeleteFalse(id).orElseThrow(() -> new ProductNotExistException(
+                "product with id: " + id + " does not exist"
+        ));
+        return productMapper.toResponse(product);
+    }
+
+    public ProductResponse update(long id, ProductRequest productRequest) {
+        Optional<Product> productById = productRepository.findByIdAndIsDeleteFalse(id);
+        Optional<CategoryProduct> categoryProduct = categoryProductRepository.findById(productRequest.getCategoryProductId());
+        if (productById.isEmpty()){
+            throw new ProductNotExistException("product with id " + id + " does not exist");
+        }
+        Product product = productById.get();
+        product.setTitle(productRequest.getTitle());
+        product.setPrice(productRequest.getPrice());
+        product.setStock(productRequest.getStock());
+        product.setDescription(productRequest.getDescription());
+        product.setPicUrl(productRequest.getPicUrl());
+        product.setCategoryProduct(categoryProduct.get());
+        Product save = productRepository.save(product);
```

Comment:
you have created productMapper, lets use it ! 
productMapper.toEntity(productRequest); /v2/